clusterloader2/pkg/measurement/util/controlled_pods_indexer.go

```diff
@@ -253,8 +257,16 @@ func (p *ControlledPodsIndexer) PodsControlledBy(obj interface{}) ([]*corev1.Pod
 	switch typeAccessor.GetKind() {
 	case "Deployment":
 		p.UIDLock.Lock()
-		for k := range p.deploymentUIDToRsUIDs[metaAccessor.GetUID()] {
-			podOwners = append(podOwners, k)
+		replicaSets, err := p.rsIndexer.ByIndex(controllerUIDIndex, string(metaAccessor.GetUID()))
+		if err != nil {
+			return nil, fmt.Errorf("failed to get replicasets controlled by %v: %w", metaAccessor.GetUID(), err)
+		}
+		for _, replicaSet := range replicaSets {
+			replicaSet, ok := replicaSet.(*appsv1.ReplicaSet)
+			if !ok {
+				return nil, fmt.Errorf("expected *appsv1.ReplicaSet; got: %T", replicaSet)
+			}
+			podOwners = append(podOwners, replicaSet.GetUID())
 		}
 		p.UIDLock.Unlock()
 	default:
```

Comment:
QQ: does the `default` path would work for bare `ReplicaSet` objects controlled by CL2?

If not, could we open an issue for that and link it here in a comment for documentation's sake?

Reply:
I think it should -- the pods set the owner directly so we should be able to get them the same way we get for deployment (deployments resolves to resplicasets, replicasets to pods)